### ISSUE TITLE
Cannot sock-shop in Kubernetes 1.16 version

### ISSUE THREAD

**Issue by karthicobla** [microservices-demo/microservices-demo#802]
/microservices-demo/deploy/kubernetes$ kubectl apply -f complete-demo.yaml --dry-run 
service/carts-db created (dry run)
service/carts created (dry run)
service/catalogue-db created (dry run)
service/catalogue created (dry run)
service/front-end created (dry run)
service/orders-db created (dry run)
service/orders created (dry run)
service/payment created (dry run)
service/queue-master created (dry run)
service/rabbitmq created (dry run)
service/shipping created (dry run)
service/user-db created (dry run)
service/user created (dry run)
unable to recognize "complete-demo.yaml": no matches for kind "Deployment" in version "extensions/v1beta1"
unable to recognize "complete-demo.yaml": no matches for kind "Deployment" in version "extensions/v1beta1"
unable to recognize "complete-demo.yaml": no matches for kind "Deployment" in version "extensions/v1beta1"
unable to recognize "complete-demo.yaml": no matches for kind "Deployment" in version "extensions/v1beta1"
unable to recognize "complete-demo.yaml": no matches for kind "Deployment" in version "extensions/v1beta1"
unable to recognize "complete-demo.yaml": no matches for kind "Deployment" in version "extensions/v1beta1"
unable to recognize "complete-demo.yaml": no matches for kind "Deployment" in version "extensions/v1beta1"
unable to recognize "complete-demo.yaml": no matches for kind "Deployment" in version "extensions/v1beta1"
unable to recognize "complete-demo.yaml": no matches for kind "Deployment" in version "extensions/v1beta1"
unable to recognize "complete-demo.yaml": no matches for kind "Deployment" in version "extensions/v1beta1"
unable to recognize "complete-demo.yaml": no matches for kind "Deployment" in version "extensions/v1beta1"
unable to recognize "complete-demo.yaml": no matches for kind "Deployment" in version "extensions/v1beta1"
unable to recognize "complete-demo.yaml": no matches for kind "Deployment" in version "extensions/v1beta1"

**Comment by highload-expert** [comment]
```
cd microservices-demo/deploy/kubernetes 
kubectl create namespace sock-shop
kubectl convert -f . | kubectl create -f -
watch kubectl get pods -n sock-shop
# wait for all in status Running
kubectl describe svc front-end -n sock-shop
```

**Comment by maxkezhang** [comment]
change appVersion from "extensions/v1beta1" to "app/v1" may fix it.

**Comment by mimor** [comment]
As the API extensions/v1beta1 has some deprecations used in the manifest files. 
For a complete list:  https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/

If you are running this in minikube, you can specify the version of the kubernetes you want to run. eg:
 
` minikube start --memory 8192 --cpus 4 --kubernetes-version v1.14.10`

**Comment by jl0x61** [comment]
> change appVersion from "extensions/v1beta1" to "app/v1" may fix it.

I tried to replace "extensions/v1beta" by "app/v1", but error messages were still there:
```
Error from server (Invalid): error when creating "complete-demo.yaml": Deployment.apps "carts-db" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"name":"carts-db"}: `selector` does not match template `labels`]
Error from server (Invalid): error when creating "complete-demo.yaml": Deployment.apps "carts" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"name":"carts"}: `selector` does not match template `labels`]
Error from server (Invalid): error when creating "complete-demo.yaml": Deployment.apps "catalogue-db" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"name":"catalogue-db"}: `selector` does not match template `labels`]
Error from server (Invalid): error when creating "complete-demo.yaml": Deployment.apps "catalogue" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"name":"catalogue"}: `selector` does not match template `labels`]
Error from server (Invalid): error when creating "complete-demo.yaml": Deployment.apps "front-end" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"name":"front-end"}: `selector` does not match template `labels`]
Error from server (Invalid): error when applying patch:
{"metadata":{"annotations":{"kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"v1\",\"kind\":\"Service\",\"metadata\":{\"annotations\":{},\"labels\":{\"name\":\"front-end\"},\"name\":\"front-end\",\"namespace\":\"sock-shop\"},\"spec\":{\"ports\":[{\"nodePort\":30001,\"port\":80,\"targetPort\":8079}],\"selector\":{\"name\":\"front-end\"},\"type\":\"NodePort\"}}\n"}},"spec":{"$setElementOrder/ports":[{"port":80}],"ports":[{"nodePort":30001,"port":80}]}}
to:
Resource: "/v1, Resource=services", GroupVersionKind: "/v1, Kind=Service"
Name: "front-end", Namespace: "sock-shop"
Object: &{map["apiVersion":"v1" "kind":"Service" "metadata":map["annotations":map["kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"v1\",\"kind\":\"Service\",\"metadata\":{\"annotations\":{},\"labels\":{\"name\":\"front-end\"},\"name\":\"front-end\",\"namespace\":\"sock-shop\"},\"spec\":{\"ports\":[{\"nodePort\":30233,\"port\":80,\"targetPort\":8079}],\"selector\":{\"name\":\"front-end\"},\"type\":\"NodePort\"}}\n"] "creationTimestamp":"2020-03-25T07:14:11Z" "labels":map["name":"front-end"] "name":"front-end" "namespace":"sock-shop" "resourceVersion":"7107066" "selfLink":"/api/v1/namespaces/sock-shop/services/front-end" "uid":"3be3e23a-10f9-480b-8265-433fff465262"] "spec":map["clusterIP":"10.68.177.167" "externalTrafficPolicy":"Cluster" "ports":[map["nodePort":'\u7619' "port":'P' "protocol":"TCP" "targetPort":'\u1f8f']] "selector":map["name":"front-end"] "sessionAffinity":"None" "type":"NodePort"] "status":map["loadBalancer":map[]]]}
for: "complete-demo.yaml": Service "front-end" is invalid: spec.ports[0].nodePort: Invalid value: 30001: provided port is already allocated
Error from server (Invalid): error when creating "complete-demo.yaml": Deployment.apps "orders-db" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"name":"orders-db"}: `selector` does not match template `labels`]
Error from server (Invalid): error when creating "complete-demo.yaml": Deployment.apps "orders" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"name":"orders"}: `selector` does not match template `labels`]
Error from server (Invalid): error when creating "complete-demo.yaml": Deployment.apps "payment" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"name":"payment"}: `selector` does not match template `labels`]
Error from server (Invalid): error when creating "complete-demo.yaml": Deployment.apps "queue-master" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"name":"queue-master"}: `selector` does not match template `labels`]
Error from server (Invalid): error when creating "complete-demo.yaml": Deployment.apps "rabbitmq" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"name":"rabbitmq"}: `selector` does not match template `labels`]
Error from server (Invalid): error when creating "complete-demo.yaml": Deployment.apps "shipping" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"name":"shipping"}: `selector` does not match template `labels`]
Error from server (Invalid): error when creating "complete-demo.yaml": Deployment.apps "user-db" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"name":"user-db"}: `selector` does not match template `labels`]
Error from server (Invalid): error when creating "complete-demo.yaml": Deployment.apps "user" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"name":"user"}: `selector` does not match template `labels`]
```
Do you know how to solve this problem? Thank you!

**Comment by kethireddysudhakar** [comment]
> > change appVersion from "extensions/v1beta1" to "app/v1" may fix it.
> 
> I tried to replace "extensions/v1beta" by "app/v1", but error messages were still there:
> 
> ```
> Error from server (Invalid): error when creating "complete-demo.yaml": Deployment.apps "carts-db" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"name":"carts-db"}: `selector` does not match template `labels`]
> Error from server (Invalid): error when creating "complete-demo.yaml": Deployment.apps "carts" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"name":"carts"}: `selector` does not match template `labels`]
> Error from server (Invalid): error when creating "complete-demo.yaml": Deployment.apps "catalogue-db" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"name":"catalogue-db"}: `selector` does not match template `labels`]
> Error from server (Invalid): error when creating "complete-demo.yaml": Deployment.apps "catalogue" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"name":"catalogue"}: `selector` does not match template `labels`]
> Error from server (Invalid): error when creating "complete-demo.yaml": Deployment.apps "front-end" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"name":"front-end"}: `selector` does not match template `labels`]
> Error from server (Invalid): error when applying patch:
> {"metadata":{"annotations":{"kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"v1\",\"kind\":\"Service\",\"metadata\":{\"annotations\":{},\"labels\":{\"name\":\"front-end\"},\"name\":\"front-end\",\"namespace\":\"sock-shop\"},\"spec\":{\"ports\":[{\"nodePort\":30001,\"port\":80,\"targetPort\":8079}],\"selector\":{\"name\":\"front-end\"},\"type\":\"NodePort\"}}\n"}},"spec":{"$setElementOrder/ports":[{"port":80}],"ports":[{"nodePort":30001,"port":80}]}}
> to:
> Resource: "/v1, Resource=services", GroupVersionKind: "/v1, Kind=Service"
> Name: "front-end", Namespace: "sock-shop"
> Object: &{map["apiVersion":"v1" "kind":"Service" "metadata":map["annotations":map["kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"v1\",\"kind\":\"Service\",\"metadata\":{\"annotations\":{},\"labels\":{\"name\":\"front-end\"},\"name\":\"front-end\",\"namespace\":\"sock-shop\"},\"spec\":{\"ports\":[{\"nodePort\":30233,\"port\":80,\"targetPort\":8079}],\"selector\":{\"name\":\"front-end\"},\"type\":\"NodePort\"}}\n"] "creationTimestamp":"2020-03-25T07:14:11Z" "labels":map["name":"front-end"] "name":"front-end" "namespace":"sock-shop" "resourceVersion":"7107066" "selfLink":"/api/v1/namespaces/sock-shop/services/front-end" "uid":"3be3e23a-10f9-480b-8265-433fff465262"] "spec":map["clusterIP":"10.68.177.167" "externalTrafficPolicy":"Cluster" "ports":[map["nodePort":'\u7619' "port":'P' "protocol":"TCP" "targetPort":'\u1f8f']] "selector":map["name":"front-end"] "sessionAffinity":"None" "type":"NodePort"] "status":map["loadBalancer":map[]]]}
> for: "complete-demo.yaml": Service "front-end" is invalid: spec.ports[0].nodePort: Invalid value: 30001: provided port is already allocated
> Error from server (Invalid): error when creating "complete-demo.yaml": Deployment.apps "orders-db" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"name":"orders-db"}: `selector` does not match template `labels`]
> Error from server (Invalid): error when creating "complete-demo.yaml": Deployment.apps "orders" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"name":"orders"}: `selector` does not match template `labels`]
> Error from server (Invalid): error when creating "complete-demo.yaml": Deployment.apps "payment" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"name":"payment"}: `selector` does not match template `labels`]
> Error from server (Invalid): error when creating "complete-demo.yaml": Deployment.apps "queue-master" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"name":"queue-master"}: `selector` does not match template `labels`]
> Error from server (Invalid): error when creating "complete-demo.yaml": Deployment.apps "rabbitmq" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"name":"rabbitmq"}: `selector` does not match template `labels`]
> Error from server (Invalid): error when creating "complete-demo.yaml": Deployment.apps "shipping" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"name":"shipping"}: `selector` does not match template `labels`]
> Error from server (Invalid): error when creating "complete-demo.yaml": Deployment.apps "user-db" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"name":"user-db"}: `selector` does not match template `labels`]
> Error from server (Invalid): error when creating "complete-demo.yaml": Deployment.apps "user" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"name":"user"}: `selector` does not match template `labels`]
> ```
> 
> Do you know how to solve this problem? Thank you!

you need to make couple more changes. i am adding the sample code for that change, need to selector object and rename the labels from name to app.

apiVersion: apps/v1
kind: Deployment
metadata:
  name: carts-db
  labels:
    app: carts-db
  namespace: sock-shop
spec:
  replicas: 1
  selector:
    matchLabels:
      app: carts-db
  template:
    metadata:
      labels:
        app: carts-db

**Comment by RubyKoviar** [comment]
Sorry for the long comment...
I changed the yaml to work with k8s 1.16 api changes
you can copy it and save it as the new complete-deploy.yaml , it works for me with GKE 1.16

```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: carts-db
  labels:
    name: carts-db
  namespace: sock-shop
spec:
  selector:
    matchLabels:
      name: carts-db
  replicas: 1
  template:
    metadata:
      labels:
        name: carts-db
    spec:
      containers:
      - name: carts-db
        image: mongo
        ports:
        - name: mongo
          containerPort: 27017
        securityContext:
          capabilities:
            drop:
              - all
            add:
              - CHOWN
              - SETGID
              - SETUID
          readOnlyRootFilesystem: true
        volumeMounts:
        - mountPath: /tmp
          name: tmp-volume
      volumes:
        - name: tmp-volume
          emptyDir:
            medium: Memory
      nodeSelector:
        beta.kubernetes.io/os: linux
---
apiVersion: v1
kind: Service
metadata:
  name: carts-db
  labels:
    name: carts-db
  namespace: sock-shop
spec:
  ports:
    # the port that this service should serve on
  - port: 27017
    targetPort: 27017
  selector:
    name: carts-db
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: carts
  labels:
    name: carts
  namespace: sock-shop
spec:
  selector:
    matchLabels:
      name: carts
  replicas: 1
  template:
    metadata:
      labels:
        name: carts
    spec:
      containers:
      - name: carts
        image: weaveworksdemos/carts:0.4.8
        ports:
         - containerPort: 80
        env:
         - name: ZIPKIN
           value: zipkin.jaeger.svc.cluster.local
         - name: JAVA_OPTS
           value: -Xms64m -Xmx128m -XX:PermSize=32m -XX:MaxPermSize=64m -XX:+UseG1GC -Djava.security.egd=file:/dev/urandom
        securityContext:
          runAsNonRoot: true
          runAsUser: 10001
          capabilities:
            drop:
              - all
            add:
              - NET_BIND_SERVICE
          readOnlyRootFilesystem: true
        volumeMounts:
        - mountPath: /tmp
          name: tmp-volume
      volumes:
        - name: tmp-volume
          emptyDir:
            medium: Memory
      nodeSelector:
        beta.kubernetes.io/os: linux
---
apiVersion: v1
kind: Service
metadata:
  name: carts
  labels:
    name: carts
  namespace: sock-shop
spec:
  ports:
    # the port that this service should serve on
  - port: 80
    targetPort: 80
  selector:
    name: carts
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: catalogue-db
  labels:
    name: catalogue-db
  namespace: sock-shop
spec:
  selector:
    matchLabels:
      name: catalogue-db
  replicas: 1
  template:
    metadata:
      labels:
        name: catalogue-db
    spec:
      containers:
      - name: catalogue-db
        image: weaveworksdemos/catalogue-db:0.3.0
        env:
          - name: MYSQL_ROOT_PASSWORD
            value: fake_password
          - name: MYSQL_DATABASE
            value: socksdb
        ports:
        - name: mysql
          containerPort: 3306
      nodeSelector:
        beta.kubernetes.io/os: linux
---
apiVersion: v1
kind: Service
metadata:
  name: catalogue-db
  labels:
    name: catalogue-db
  namespace: sock-shop
spec:
  ports:
    # the port that this service should serve on
  - port: 3306
    targetPort: 3306
  selector:
    name: catalogue-db
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: catalogue
  labels:
    name: catalogue
  namespace: sock-shop
spec:
  selector:
    matchLabels:
      name: catalogue
  replicas: 1
  template:
    metadata:
      labels:
        name: catalogue
    spec:
      containers:
      - name: catalogue
        image: weaveworksdemos/catalogue:0.3.5
        ports:
        - containerPort: 80
        securityContext:
          runAsNonRoot: true
          runAsUser: 10001
          capabilities:
            drop:
              - all
            add:
              - NET_BIND_SERVICE
          readOnlyRootFilesystem: true
      nodeSelector:
        beta.kubernetes.io/os: linux
---
apiVersion: v1
kind: Service
metadata:
  name: catalogue
  labels:
    name: catalogue
  namespace: sock-shop
spec:
  ports:
    # the port that this service should serve on
  - port: 80
    targetPort: 80
  selector:
    name: catalogue
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: front-end
  namespace: sock-shop
spec:
  selector:
    matchLabels:
      name: front-end
  replicas: 1
  template:
    metadata:
      labels:
        name: front-end
    spec:
      containers:
      - name: front-end
        image: weaveworksdemos/front-end:0.3.12
        resources:
          requests:
            cpu: 100m
            memory: 100Mi
        ports:
        - containerPort: 8079
        securityContext:
          runAsNonRoot: true
          runAsUser: 10001
          capabilities:
            drop:
              - all
          readOnlyRootFilesystem: true
      nodeSelector:
        beta.kubernetes.io/os: linux
---
apiVersion: v1
kind: Service
metadata:
  name: front-end
  labels:
    name: front-end
  namespace: sock-shop
spec:
  type: NodePort
  ports:
  - port: 80
    targetPort: 8079
    nodePort: 30001
  selector:
    name: front-end
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: orders-db
  labels:
    name: orders-db
  namespace: sock-shop
spec:
  selector:
    matchLabels:
      name: orders-db
  replicas: 1
  template:
    metadata:
      labels:
        name: orders-db
    spec:
      containers:
      - name: orders-db
        image: mongo
        ports:
        - name: mongo
          containerPort: 27017
        securityContext:
          capabilities:
            drop:
              - all
            add:
              - CHOWN
              - SETGID
              - SETUID
          readOnlyRootFilesystem: true
        volumeMounts:
        - mountPath: /tmp
          name: tmp-volume
      volumes:
        - name: tmp-volume
          emptyDir:
            medium: Memory
      nodeSelector:
        beta.kubernetes.io/os: linux
---
apiVersion: v1
kind: Service
metadata:
  name: orders-db
  labels:
    name: orders-db
  namespace: sock-shop
spec:
  ports:
    # the port that this service should serve on
  - port: 27017
    targetPort: 27017
  selector:
    name: orders-db
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: orders
  labels:
    name: orders
  namespace: sock-shop
spec:
  selector:
    matchLabels:
      name: orders
  replicas: 1
  template:
    metadata:
      labels:
        name: orders
    spec:
      containers:
      - name: orders
        image: weaveworksdemos/orders:0.4.7
        env:
         - name: ZIPKIN
           value: zipkin.jaeger.svc.cluster.local
         - name: JAVA_OPTS
           value: -Xms64m -Xmx128m -XX:PermSize=32m -XX:MaxPermSize=64m -XX:+UseG1GC -Djava.security.egd=file:/dev/urandom
        ports:
        - containerPort: 80
        securityContext:
          runAsNonRoot: true
          runAsUser: 10001
          capabilities:
            drop:
              - all
            add:
              - NET_BIND_SERVICE
          readOnlyRootFilesystem: true
        volumeMounts:
        - mountPath: /tmp
          name: tmp-volume
      volumes:
        - name: tmp-volume
          emptyDir:
            medium: Memory
      nodeSelector:
        beta.kubernetes.io/os: linux
---
apiVersion: v1
kind: Service
metadata:
  name: orders
  labels:
    name: orders
  namespace: sock-shop
spec:
  ports:
    # the port that this service should serve on
  - port: 80
    targetPort: 80
  selector:
    name: orders
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: payment
  labels:
    name: payment
  namespace: sock-shop
spec:
  selector:
    matchLabels:
      name: payment
  replicas: 1
  template:
    metadata:
      labels:
        name: payment
    spec:
      containers:
      - name: payment
        image: weaveworksdemos/payment:0.4.3
        ports:
        - containerPort: 80
        securityContext:
          runAsNonRoot: true
          runAsUser: 10001
          capabilities:
            drop:
              - all
            add:
              - NET_BIND_SERVICE
          readOnlyRootFilesystem: true
      nodeSelector:
        beta.kubernetes.io/os: linux
---
apiVersion: v1
kind: Service
metadata:
  name: payment
  labels:
    name: payment
  namespace: sock-shop
spec:
  ports:
    # the port that this service should serve on
  - port: 80
    targetPort: 80
  selector:
    name: payment
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: queue-master
  labels:
    name: queue-master
  namespace: sock-shop
spec:
  selector:
    matchLabels:
      name: queue-master
  replicas: 1
  template:
    metadata:
      labels:
        name: queue-master
    spec:
      containers:
      - name: queue-master
        image: weaveworksdemos/queue-master:0.3.1
        ports:
        - containerPort: 80
      nodeSelector:
        beta.kubernetes.io/os: linux
---
apiVersion: v1
kind: Service
metadata:
  name: queue-master
  labels:
    name: queue-master
  annotations:
    prometheus.io/path: "/prometheus"
  namespace: sock-shop
spec:
  ports:
    # the port that this service should serve on
  - port: 80
    targetPort: 80
  selector:
    name: queue-master
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: rabbitmq
  labels:
    name: rabbitmq
  namespace: sock-shop
spec:
  selector:
    matchLabels:
      name: rabbitmq
  replicas: 1
  template:
    metadata:
      labels:
        name: rabbitmq
    spec:
      containers:
      - name: rabbitmq
        image: rabbitmq:3.6.8
        ports:
        - containerPort: 5672
        securityContext:
          capabilities:
            drop:
              - all
            add:
              - CHOWN
              - SETGID
              - SETUID
              - DAC_OVERRIDE
          readOnlyRootFilesystem: true
      nodeSelector:
        beta.kubernetes.io/os: linux
---
apiVersion: v1
kind: Service
metadata:
  name: rabbitmq
  labels:
    name: rabbitmq
  namespace: sock-shop
spec:
  ports:
    # the port that this service should serve on
  - port: 5672
    targetPort: 5672
  selector:
    name: rabbitmq
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: shipping
  labels:
    name: shipping
  namespace: sock-shop
spec:
  selector:
    matchLabels:
      name: shipping
  replicas: 1
  template:
    metadata:
      labels:
        name: shipping
    spec:
      containers:
      - name: shipping
        image: weaveworksdemos/shipping:0.4.8
        env:
         - name: ZIPKIN
           value: zipkin.jaeger.svc.cluster.local
         - name: JAVA_OPTS
           value: -Xms64m -Xmx128m -XX:PermSize=32m -XX:MaxPermSize=64m -XX:+UseG1GC -Djava.security.egd=file:/dev/urandom
        ports:
        - containerPort: 80
        securityContext:
          runAsNonRoot: true
          runAsUser: 10001
          capabilities:
            drop:
              - all
            add:
              - NET_BIND_SERVICE
          readOnlyRootFilesystem: true
        volumeMounts:
        - mountPath: /tmp
          name: tmp-volume
      volumes:
        - name: tmp-volume
          emptyDir:
            medium: Memory
      nodeSelector:
        beta.kubernetes.io/os: linux
---
apiVersion: v1
kind: Service
metadata:
  name: shipping
  labels:
    name: shipping
  namespace: sock-shop
spec:
  ports:
    # the port that this service should serve on
  - port: 80
    targetPort: 80
  selector:
    name: shipping
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: user-db
  labels:
    name: user-db
  namespace: sock-shop
spec:
  selector:
    matchLabels:
      name: user-db
  replicas: 1
  template:
    metadata:
      labels:
        name: user-db
    spec:
      containers:
      - name: user-db
        image: weaveworksdemos/user-db:0.4.0
        ports:
        - name: mongo
          containerPort: 27017
        securityContext:
          capabilities:
            drop:
              - all
            add:
              - CHOWN
              - SETGID
              - SETUID
          readOnlyRootFilesystem: true
        volumeMounts:
        - mountPath: /tmp
          name: tmp-volume
      volumes:
        - name: tmp-volume
          emptyDir:
            medium: Memory
      nodeSelector:
        beta.kubernetes.io/os: linux
---
apiVersion: v1
kind: Service
metadata:
  name: user-db
  labels:
    name: user-db
  namespace: sock-shop
spec:
  ports:
    # the port that this service should serve on
  - port: 27017
    targetPort: 27017
  selector:
    name: user-db
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: user
  labels:
    name: user
  namespace: sock-shop
spec:
  selector:
    matchLabels:
      name: user
  replicas: 1
  template:
    metadata:
      labels:
        name: user
    spec:
      containers:
      - name: user
        image: weaveworksdemos/user:0.4.7
        ports:
        - containerPort: 80
        env:
        - name: MONGO_HOST
          value: user-db:27017
        securityContext:
          runAsNonRoot: true
          runAsUser: 10001
          capabilities:
            drop:
              - all
            add:
              - NET_BIND_SERVICE
          readOnlyRootFilesystem: true
      nodeSelector:
        beta.kubernetes.io/os: linux
---
apiVersion: v1
kind: Service
metadata:
  name: user
  labels:
    name: user
  namespace: sock-shop
spec:
  ports:
    # the port that this service should serve on
  - port: 80
    targetPort: 80
  selector:
    name: user

```

**Comment by bonen02** [comment]
> 
> 
> ```
> cd microservices-demo/deploy/kubernetes 
> kubectl create namespace sock-shop
> kubectl convert -f . | kubectl create -f -
> watch kubectl get pods -n sock-shop
> # wait for all in status Running
> kubectl describe svc front-end -n sock-shop
> ```

This. Works also with Kubernetes 1.18.5 
Thanks a lot!

**Comment by sureshreddya** [comment]
changind deployment

**Comment by angadrandhawa** [comment]
> ```
> cd microservices-demo/deploy/kubernetes 
> kubectl create namespace sock-shop
> kubectl convert -f . | kubectl create -f -
> watch kubectl get pods -n sock-shop
> # wait for all in status Running
> kubectl describe svc front-end -n sock-shop
> ```

starting 1.19 version `kubectl convert` has been deprecated. I read somewhere that it is being converted to a plugin, but at the moment i cannot even find that plugin.

**Comment by rodrmartinez** [comment]
this issues has been solved by #809
